app/configurator/components/field.tsx

```diff
@@ -293,122 +296,151 @@ export const MetaInputField = ({
-          value: {
-            dimensionIri,
-            value,
-            allValues,
-          },
-        });
-      } else {
-        dispatch({
-          type: "CHART_CONFIG_FILTER_REMOVE_MULTI",
-          value: {
-            dimensionIri,
-            value,
-            allValues,
-          },
-        });
-      }
-      // Call onChange prop
-      onChange?.();
-    },
-    [dispatch, dimensionIri, allValues, value, onChange, checkAction]
-  );
-
-  const updateColor = useCallback(
+const useMultiFilterColorPicker = (
+  value: string,
+  dimensionIri: string,
+  colorConfigPath?: string
+) => {
+  const [configuratorState, dispatch] = useConfiguratorState();
+  const { isFilterActive, selectionState } = useMultiFilterContext();
+  const { activeField } = configuratorState;
+  const onChange = useCallback(
     (color: string) => {
-      if (state.activeField) {
+      if (activeField) {
         dispatch({
           type: "CHART_COLOR_CHANGED",
           value: {
-            field: state.activeField,
+            field: activeField,
             colorConfigPath,
             color,
             value,
           },
         });
       }
     },
-    [colorConfigPath, dispatch, state.activeField, value]
+
+    [colorConfigPath, dispatch, activeField, value]
+  );
+
+  const path = colorConfigPath ? `${colorConfigPath}.` : "";
+  const chartConfig =
+    configuratorState.state === "CONFIGURING_CHART"
+      ? configuratorState.chartConfig
+      : null;
+
+  const color = chartConfig
+    ? get(
+        chartConfig,
+        `fields["${activeField}"].${path}colorMapping["${value}"]`
+      )
+    : null;
+
+  const palette = useMemo(() => {
+    if (!chartConfig) {
+      return [];
+    }
+    return getPalette(
+      get(
+        chartConfig,
+        `fields["${activeField}"].${colorConfigPath ?? ""}.palette`
+      )
+    );
+  }, [chartConfig, colorConfigPath, activeField]);
+
+  const checkedState =
+    configuratorState.state === "CONFIGURING_CHART"
+      ? isMultiFilterFieldChecked(
+          configuratorState.chartConfig,
+          dimensionIri,
+          value
+        )
+      : null;
+
+  return useMemo(
+    () => ({
+      color,
+      palette,
+      onChange,
+      checked:
+        selectionState === "ALL_SELECTED"
+          ? true
+          : selectionState === "SOME_SELECTED"
+          ? !!isFilterActive.has(value)
+          : undefined && checkedState,
+    }),
+    [
+      color,
+      palette,
+      onChange,
+      selectionState,
+      isFilterActive,
+      value,
+      checkedState,
+    ]
+  );
+};
+
+export const MultiFilterFieldColorPicker = ({
+  colorConfigPath,
+  value,
+  dimensionIri,
+}: {
+  colorConfigPath?: string;
+  value: string;
+  dimensionIri: string;
+  checked?: boolean;
+}) => {
+  const { color, checked, palette, onChange } = useMultiFilterColorPicker(
+    value,
+    dimensionIri,
+    colorConfigPath
+  );
+
+  return color && checked ? (
```

Comment:
Nit: color && checked && (...)?

Reply:
Yes it can be done but I do not think it changes anything though :) I think I prefer the ternary form because it's easier to add the falsy case.